### PR TITLE
Add installed themes and WooCommerce support to theme browser

### LINKED ISSUE #2494: Onboarding: Add already installed themes to theme step
The onboarding theme step in https://github.com/woocommerce/woocommerce-admin/pull/2493 also shows pre-installed themes as options in the theme browser. 

This issue involves:
* Adding pre-installed themes to the list of themes shown.
* Checking for theme support for woocommerce and adding a boolean flag to each theme.
* Adding theme (lack of) support icon and updating prices on front-end.
* Adding the currently active theme slug to the onboarding data.
* Potentially sorting themes.

Screen Shot 2019-06-24 at 6 39 02 PM

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #2494

Adds installed themes to the theme browser as well as WC support error icons if the theme doesn't support WC.

### Questions/discussion

I initially looked at using `get_theme_support()` but this only works for the active theme.  Trying to include all themes and artificially activate this seems like it might cause a lot of trouble.

The current solution uses a regex match.  It's not perfect, but I think might do a decent job of detecting support.  It also puts the installed themes in the transient to prevent potential performance issues.

### Screenshots
Screen Shot 2019-06-25 at 6 11 38 PM

### Detailed test instructions:

1. Get a local woocommerce.com test site running and use [this branch](https://github.com/Automattic/woocommerce.com/pull/6035).
2. Install Storefront or other wc supported themes.
3. Visit `wp-admin/admin.php?page=wc-admin#/?step=theme` in your local WP install.
4.  Make sure already installed themes show up in the list and are styled correctly (no "Live Demo" option and price/status line should state `Installed` or `Currently active theme`).
5.  Make sure Storefront does not show a lack of support error icon.

### Changelog Note:

Enhancement: Add installed themes to theme browser during onboarding.

### CODE REVIEW

**Comment by rrennick** [comment]
@joshuatf I'm getting a 500 error in my local install

```
[25-Jun-2019 13:01:13 UTC] PHP Fatal error:  Uncaught Error: Cannot use object of type WP_Error as array in /Users/ronrennick/Sites/ron/wp-content/plugins/woocommerce-admin/includes/features/onboarding/class-wc-admin-onboarding.php:127
Stack trace:
#0 /Users/ronrennick/Sites/ron/wp-content/plugins/woocommerce-admin/includes/features/onboarding/class-wc-admin-onboarding.php(182): WC_Admin_Onboarding::get_themes()
#1 /Users/ronrennick/Sites/ron/wp-includes/class-wp-hook.php(286): WC_Admin_Onboarding->component_settings(Array)
#2 /Users/ronrennick/Sites/ron/wp-includes/plugin.php(208): WP_Hook->apply_filters(Array, Array)
#3 /Users/ronrennick/Sites/ron/wp-content/plugins/woocommerce/includes/blocks/class-wc-block-library.php(279): apply_filters('woocommerce_com...', Array)
#4 /Users/ronrennick/Sites/ron/wp-includes/class-wp-hook.php(286): WC_Block_Library::print_script_settings('')
#5 /Users/ronrennick/Sites/ron/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters('', Array)
#6 /Users/ronrennick/Sites/ron/wp-includes/plugin.php(465): WP_Hook->do_acti in /Users/ronrennick/Sites/ron/wp-content/plugins/woocommerce-admin/includes/features/onboarding/class-wc-admin-onboarding.php on line 127
```

**Comment by joshuatf** [comment]
Thanks for taking a look @rrennick!  Sorry, the link was missing above, but can you confirm that you're running on [this branch](https://github.com/Automattic/woocommerce.com/pull/6035) for your local wccom test site?

**Comment by justinshreve** [comment]
> Make sure already installed themes show up in the list and are styled correctly (no "Live Demo" option and price/status line should state Installed or Currently active theme).

I wonder if we should consider showing a `preview` link for installed themes? Or would the rest of the customizer controls get confusing?

File: client/dashboard/profile-wizard/steps/theme/error.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/** @format */
+
+export default () => (
+	<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (justinshreve) [inline client/dashboard/profile-wizard/steps/theme/error.js]: I think we can use the `info` Gridicon here.

**Comment by justinshreve** [comment]
How should we handle installed themes that we are not sure of the price for? For example, I have the Avada theme installed from when I helped out in support, and it shows up under the "Free" tab, but it costs money off-site. Should we have 4 tabs? `All | Installed | Paid | Free`?

File: client/dashboard/profile-wizard/steps/theme/index.js [hunk 2]
```diff
@@ -55,33 +56,64 @@ class Theme extends Component {
 	}
 
 	renderTheme( theme ) {
-		const { image, price, slug, title } = theme;
+		const { demo, has_woocommerce_support, image, slug, title } = theme;
```
Comment (justinshreve) [inline client/dashboard/profile-wizard/steps/theme/index.js]: Heads up: `demo` is changing to `demo_url` in the API response.

Comment (joshuatf) [reply client/dashboard/profile-wizard/steps/theme/index.js]: Thanks- updated!

File: client/dashboard/profile-wizard/steps/theme/error.js (hunk 1 above)
Comment (joshuatf) [reply client/dashboard/profile-wizard/steps/theme/error.js]: Nice find!  I think this is close enough to the original icon.  @jameskoster Let me know if you see any issues with using this one instead. (part of the gridicon library we're already using)

Screen Shot 2019-06-26 at 11 06 12 AM

**Comment by joshuatf** [comment]
>Should we have 4 tabs? All | Installed | Paid | Free?

Good question.  I think adding an "Installed" tab is probably the best, but if we don't have the screen real estate to do it, I can see it going under the "Free" tab since it doesn't come at an additional cost.

>I wonder if we should consider showing a preview link for installed themes? Or would the rest of the customizer controls get confusing (or maybe we could hide those)?

I think this is a good idea, but we might want to hold it for v2.  I imagine that might add a good amount of complexity.

**Comment by joshuatf** [comment]
Also, one more thing to note, @justinshreve - it looks like your image gist still hasn't been run on wccom yet.  Not sure if that was supposed to have already happened or not:

Screen Shot 2019-06-26 at 11 06 21 AM

**Comment by justinshreve** [comment]
> Good question. I think adding an "Installed" tab is probably the best, but if we don't have the screen real estate to do it, I can see it going under the "Free" tab since it doesn't come at an additional cost.

Since it sounds like we won't be displaying all installed themes (https://github.com/woocommerce/woocommerce-admin/pull/2493#issuecomment-505779733) I think you are right and the free tab is fine. Maybe we could just hide the additional "free" under the image if it's an installed theme?

**Comment by justinshreve** [comment]
> it looks like your image gist still hasn't been run on wccom yet. Not sure if that was supposed to have already happened or not:

I was going to update the products manually instead of trying to get a `gist` ran, but it looks like there might be some caching issues or something. I'll plan on following up: p1561563265019000-slack-wc-onboarding

**Comment by jameskoster** [comment]
+1 on the free tab, for now.

If offering a preview means sending people to the Customizer I think we should avoid that. Since these themes are already installed I think it's fairly safe to assume that the user already has a good idea of how they look. Previews for the commercial themes is much more important.

**Comment by joshuatf** [comment]
Thanks for the review @justinshreve!  I'll create a follow-up issue for sorting themes and limiting to themes mentioned in https://github.com/woocommerce/woocommerce-admin/pull/2493#issuecomment-505779733.

**Pull request merged**